### PR TITLE
Port NPN and ALPN tests

### PR DESCRIPTION
**Pull request by ekasper**
Rewrite NPN and ALPN tests in the new framework. Fix a bug found by the new tests, and a second bug found while fixing the first bug. (See second commit for details.)

### CODE REVIEW

File: ssl/t1_lib.c [hunk 1]
```diff
@@ -1682,7 +1682,7 @@ static int tls1_alpn_handle_client_hello(SSL *s, PACKET *pkt, int *al)
  * al: a pointer to the alert value to send in the event of a failure.
  * returns 1 on success, 0
```
Comment (kroeckx) [inline ssl/t1_lib.c]: Could you fix that comment, after the "0"?

Comment (kroeckx) [inline ssl/t1_lib.c]: There is also a new parameter ...

Comment (ekasper) [inline ssl/t1_lib.c]: Fixed.

**Comment by richsalz** [comment]
looks good to me, but @kroeckx ?

**Comment by ekasper** [comment]
Ping...

**Comment by richsalz** [comment]
to make previous comment explicit:  +1 from me, if you don't want to hear from kurt.

**Pull request closed without merging**